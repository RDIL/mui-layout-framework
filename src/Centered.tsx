import * as React from "react"
import Grid from "@material-ui/core/Grid"

export interface CenteredProps {
    children: React.ReactElement
}

const Centered = (props: CenteredProps) => {
    return (
        <Grid container justify="center">
            {props.children}
        </Grid>
    )
}

export default Centered
