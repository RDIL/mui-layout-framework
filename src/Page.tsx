import * as React from "react"
import { Helmet } from "react-helmet"

export interface PageProps {
    title: string
    children: React.ReactElement | React.ReactElement[]
    header?: React.ReactElement
    footer?: React.ReactElement
    scripts?: string[]
}

/**
 * The component for a typical page.
 */
const Page = (props: PageProps) => {
    return (
        <div>
            <Helmet>
                <title>{props.title}</title>
                {props.scripts && props.scripts.length > 0
                    ? props.scripts.map((url) => (
                          <script defer src={url} key={url} />
                      ))
                    : null}
            </Helmet>
            {props.header || null}
            <main>{props.children}</main>
            {props.footer || null}
        </div>
    )
}

export default Page
