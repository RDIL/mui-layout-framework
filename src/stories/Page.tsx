import React from "react"
import "./page.css"
import Page from "../Page"
import { makeStyles } from "@material-ui/core/styles"
import { AppBar, Toolbar, Typography } from "@material-ui/core"

export const StoryPageMain = () => (
    <Page title={"Storybook Example Content Page"}>
        <h2>Pages in Storybook</h2>
        <p>
            This is a page. It is essentially a wrapper for your content.
        </p>
        <p>
            Just put whatever you want in here.
        </p>
    </Page>
)

const useStructuredPageStyles = makeStyles(() => ({
    flex: {
        flexGrow: 1,
    },
}))

export const StructuredMuiPage = () => {
    const styles = useStructuredPageStyles()

    return (
        <Page
            title={"Timmmy's Animal Rescue"}
            header={
                <AppBar>
                    <Toolbar className={styles.flex}>
                        <Typography variant="h6">Timmy's Animal Rescue</Typography>
                    </Toolbar>
                </AppBar>
            }
        >
            <p>
                This page uses a header and footer, built off Material-UI!
            </p>
        </Page>
    )
}
