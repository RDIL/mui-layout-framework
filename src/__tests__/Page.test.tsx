import React from "react"
import ReactDOM from "react-dom/server"
import Page from "../Page"
import Centered from "../Centered"
import { AppBar, Toolbar, Typography } from "@material-ui/core"

describe("Page component", () => {
    it("can render a basic page", () => {
        expect(
            ReactDOM.renderToString(
                <Page title="Tim's Animal Rescue">
                    <Centered>
                        <Typography variant="h2">
                            Tim's Animal Rescue
                        </Typography>
                    </Centered>
                </Page>
            )
        ).toMatchSnapshot()
    })

    it("can render a page with scripts", () => {
        // right now we can't check that its actually been applied, so lets just pray it works
        expect(
            ReactDOM.renderToString(
                <Page
                    title="Tim's Animal Rescue"
                    scripts={["https://example.com/example.js"]}
                >
                    <Centered>
                        <Typography variant="h2">
                            Tim's Animal Rescue
                        </Typography>
                    </Centered>
                </Page>
            )
        ).toMatchSnapshot()
    })

    it("can render a page a header and footer", () => {
        const Header = () => (
            <AppBar>
                <Toolbar>
                    <Typography variant="h2">My Navbar</Typography>
                </Toolbar>
            </AppBar>
        )

        const Footer = () => (
            <div>
                <Typography>some footer</Typography>
            </div>
        )

        expect(
            ReactDOM.renderToString(
                <Page
                    title="Tim's Animal Rescue"
                    header={<Header />}
                    footer={<Footer />}
                >
                    <Typography>My cool content</Typography>
                </Page>
            )
        ).toMatchSnapshot()
    })
})
