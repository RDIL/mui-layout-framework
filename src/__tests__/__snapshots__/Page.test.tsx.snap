// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component can render a basic page 1`] = `"<div data-reactroot=\\"\\"><main><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center\\"><h2 class=\\"MuiTypography-root MuiTypography-h2\\">Tim&#x27;s Animal Rescue</h2></div></main></div>"`;

exports[`Page component can render a page a header and footer 1`] = `"<div data-reactroot=\\"\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><h2 class=\\"MuiTypography-root MuiTypography-h2\\">My Navbar</h2></div></header><main><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center\\"><h2 class=\\"MuiTypography-root MuiTypography-h2\\">Tim&#x27;s Animal Rescue</h2></div></main><div><p class=\\"MuiTypography-root MuiTypography-body1\\">some footer</p></div></div>"`;

exports[`Page component can render a page with scripts 1`] = `"<div data-reactroot=\\"\\"><main><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center\\"><h2 class=\\"MuiTypography-root MuiTypography-h2\\">Tim&#x27;s Animal Rescue</h2></div></main></div>"`;
